CANTroller2/src/devices.h

```diff
@@ -392,6 +397,133 @@ class Sensor : public Transducer<NATIVE_T, HUMAN_T> {
     std::shared_ptr<HUMAN_T> get_filtered_value_ptr() { return _val_filt.get_ptr(); } // NOTE: should just be public?
 };
 
+// Base class for sensors which communicate using i2c.
+// NOTE: this is a strange type of Sensor, it's not really a Transducer but it does do filtering. maybe we should rethink the hierarchy a little?
```

Comment:
In another diff, let's move Param to utils.h
We could also move EMA stuff to utils.h
If the EMA calculation involves maintaining state (like previous averages, etc.) over multiple calls, or if it has multiple related operations/methods, let's make it a class - if it's simple enough though, let's keep it as a util function?

Reply:
yeah i've got a plan for how to do this, but i wanna make it in its own PR so it doesn't cloud up things